app/routes.py

```diff
@@ -1 +1,212 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, jsonify, make_response, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+import datetime
+from dotenv import load_dotenv
+import requests
+import os
+
+load_dotenv()
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# validation helper function
+
+def validate_item(model, item_id):
+    try:
+        item_id = int(item_id)
+    except ValueError:
+        abort(make_response({"msg": f"{model.__name__} {item_id} invalid"}, 400))
+    
+    item = model.query.get(item_id)
+    
+    if not item:
+        abort(make_response({"msg": f"{model.__name__} {item_id} not found"}, 404))
+    
+    return item
+
+# Task model routes
+
+@task_bp.route("", methods=["POST"])
+def add_task():
+    request_body = request.get_json()
+    try:
+        new_task = Task.from_dict(request_body)
+    except KeyError:
+        return {
+            "details": "Invalid data"
+        }, 400
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+@task_bp.route("", methods=["GET"])
+def get_tasks():
+    response = []
+
+    title_query = request.args.get("title")
+    sort_query = request.args.get("sort")
+
+    if title_query:
+        tasks = Task.query.filter_by(title=title_query)
+    elif sort_query == "asc":
+        tasks = Task.query.order_by(Task.title)
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+    
+    #refactor to do list comprehension
+    for task in tasks: 
+        response.append(task.to_dict())
+
+    
+    print(tasks)
+
+    return (jsonify(response), 200)
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_item(Task, task_id)
+
+    if task.goal_id:
+        return {"task":task.to_dict_with_goal()}, 200
+
+    return {"task":task.to_dict()}, 200 
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_item(Task,task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
I'd recommend wrapping these lines in a try-except block so that you can return an error message if the user is missing some of the required data